error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/MemberOrdering.java

```diff
@@ -0,0 +1,196 @@
+    for (int i = 0; i < memberWithComments.size(); i++) {
+      Tree originalMember = memberWithComments.get(i).member();
+      MemberWithComments correct = sortedMembersWithComments.get(i);
+      /* Technically this check is not necessary, but it avoids redundant replacements. */
+      if (!originalMember.equals(correct.member())) {
+        String replacement =
+            Stream.concat(
+                    correct.comments().stream().map(Tokens.Comment::getText),
+                    Stream.of(state.getSourceForNode(correct.member())))
+                .collect(joining("\n"));
+        fix.merge(
+            SuggestedFixes.replaceIncludingComments(
+                TreePath.getPath(state.getPath(), originalMember), replacement, state));
+      }
+    }
+    return fix.build();
+  }
+
+  // XXX: Work around that `ErrorProneTokens.getTokens(memberSrc, ctx)` returns tokens not
+  //  containing the member's comments.
+  /** Returns the class' members with their comments. */
+  private static ImmutableList<MemberWithComments> getMembersWithComments(
+      ClassTree classTree, VisitorState state) {
+    List<ErrorProneToken> tokens =
+        new ArrayList<>(
+            ErrorProneTokens.getTokens(state.getSourceForNode(classTree), state.context));
+
+    ImmutableList.Builder<MemberWithComments> membersWithComments = ImmutableList.builder();
+    for (Tree member : classTree.getMembers()) {
+      ImmutableList<ErrorProneToken> memberTokens =
+          ErrorProneTokens.getTokens(state.getSourceForNode(member), state.context);
+      if (memberTokens.isEmpty() || memberTokens.get(0).kind() == Tokens.TokenKind.EOF) {
+        continue;
+      }
+
+      @Var
+      ImmutableList<ErrorProneToken> maybeCommentedMemberTokens =
+          ImmutableList.copyOf(tokens.subList(0, memberTokens.size()));
+      while (!areTokenListsMatching(memberTokens, maybeCommentedMemberTokens)) {
+        tokens.remove(0);
+        maybeCommentedMemberTokens = ImmutableList.copyOf(tokens.subList(0, memberTokens.size()));
+      }
+
+      membersWithComments.add(
+          new MemberWithComments(
+              member, ImmutableList.copyOf(maybeCommentedMemberTokens.get(0).comments())));
+    }
+    return membersWithComments.build();
+  }
+
+  /**
+   * Checks whether two lists of error-prone tokens are 'equal' without considering their comments.
+   */
+  private static boolean areTokenListsMatching(
+      ImmutableList<ErrorProneToken> tokens, ImmutableList<ErrorProneToken> memberTokens) {
+    if (tokens.size() != memberTokens.size()) {
+      return false;
+    }
+    for (int i = 0; i < tokens.size() - 1 /* EOF */; i++) {
+      if (tokens.get(i).kind() != memberTokens.get(i).kind()
+          || tokens.get(i).hasName() != memberTokens.get(i).hasName()
+          || (tokens.get(i).hasName()
+              && !tokens.get(i).name().equals(memberTokens.get(i).name()))) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
```suggestion
  /** Returns the class' members with their comments. */
  private static ImmutableList<MemberWithComments> getMembersWithComments(
      ClassTree classTree, VisitorState state) {
    List<ErrorProneToken> tokens =
        state.getOffsetTokens(
            ASTHelpers.getStartPosition(classTree), state.getEndPosition(classTree));

    ImmutableList.Builder<MemberWithComments> membersWithComments = ImmutableList.builder();
    for (Tree member : classTree.getMembers()) {
      int memberStartPos = ASTHelpers.getStartPosition(member);
      // we start and the end position of the previous token
      // this gives us a range that includes any comments
      int startPos =
          tokens.stream()
              .map(ErrorProneToken::endPos)
              .filter(i -> i < memberStartPos)
              .reduce((first, second) -> second)
              .orElse(memberStartPos);

      ImmutableList<ErrorProneToken> memberTokens =
          ImmutableList.copyOf(state.getOffsetTokens(startPos, state.getEndPosition(member)));
      if (memberTokens.isEmpty() || memberTokens.get(0).kind() == Tokens.TokenKind.EOF) {
        continue;
      }

      membersWithComments.add(
          new MemberWithComments(member, ImmutableList.copyOf(memberTokens.get(0).comments())));
    }
    return membersWithComments.build();
  }
```
`ErrorProneTokens#getTokens(String, Context)` doesn't include the comments because it uses the start position of the provided member, e.g. the position of `void`. And while comments are not tokens, they do offset the position of tokens.

Using `VisitorState#getOffsetTokens(int, int)` we're able to explicitly choose the start/end positions of the token range. If we start the range at the end position of the previous token and end it at the end of the current member, we get the same tokens, but since the range now includes the offset for any comments these will be included in the resulting `ErrorProneToken`.